jba/src/test_logs/tests_analysis.py

```diff
@@ -0,0 +1,316 @@
+        [TestDataField.CLASS_NAME.value, TestDataField.METHOD_NAME.value, TestDataField.TEST_NUMBER.value],
+        dropna=False,
+    )
+
+    fig, ax = plt.subplots()
+
+    yticks = []
+    yticklabels = []
+    for i, (unique_test_name, test_timeline) in enumerate(timeline_by_unique_test_name, start=1):
+        class_name, method_name, test_number = unique_test_name
+
+        xranges = []
+        colors = []
+        for row in test_timeline.itertuples():
+            start = getattr(row, START_COLUMN)
+            finish = getattr(row, FINISH_COLUMN)
+            duration = finish - start
+            color = _get_result_color(getattr(row, TestDataField.RESULT.value))
+
+            if duration == 0:
+                plt.plot(start, i + 0.25, marker='o', markerfacecolor=color, markeredgecolor=color, markersize=5)
+            else:
+                xranges.append((start, duration))
+                colors.append(color)
+
+        test_name = f'{class_name}.{method_name}'
+        if not pd.isna(test_number):
+            # We use explicit string concatenation here for the sake of brevity
+            test_name += f'[{int(test_number)}]'  # noqa: WPS336
+
+        yticks.append(i + 0.25)
+        yticklabels.append(test_name)
+
+        plt.broken_barh(xranges=xranges, yrange=(i, 0.5), facecolors=colors)
+
+    ax.set_yticks(yticks)
+    ax.set_yticklabels(yticklabels)
+    ax.invert_yaxis()
+
+    ax.legend(
+        handles=[
+            Patch(facecolor=FAILED_COLOR, label='Failed'),
+            Patch(facecolor=IGNORED_COLOR, label='Ignored'),
+            Patch(facecolor=PASSED_COLOR, label='Passed'),
+            Patch(facecolor=DUPLICATE_COLOR, label='Duplicate'),
+        ],
+        bbox_to_anchor=(1.02, 1),
+        borderaxespad=0,
+        loc='upper left',
+    )
+
+    left_boundary = min([tests_timeline[START_COLUMN].min(), *invalid_attempts])
+    right_boundary = max([tests_timeline[FINISH_COLUMN].max(), *invalid_attempts])
+    ax.set_xticks(range(left_boundary, right_boundary + 1))
+
+    for attempt in duplicate_attempts:
+        ax.get_xticklabels()[attempt - 1].set_color(DUPLICATE_COLOR)
+
+    st.pyplot(fig)
+
+
+def main():
+    st.title('Tests Analysis')
+
+    submissions_path = st.text_input('Submissions path:')
+    if not submissions_path:
+        st.info('You should enter the submissions path.')
+        st.stop()
+
+    submissions = read_df(submissions_path)
+
+    group = st.number_input(
+        'Group:',
+        value=submissions[SubmissionColumns.GROUP.value].min(),
+        min_value=submissions[SubmissionColumns.GROUP.value].min(),
+        max_value=submissions[SubmissionColumns.GROUP.value].max(),
+    )
+
+    group_submissions = submissions[submissions[SubmissionColumns.GROUP.value] == group].reset_index(drop=True)
+
+    if group_submissions[EduColumnName.TASK_TYPE.value].iloc[0] == EduTaskType.THEORY.value:
+        st.info("It's a theory group. Please choose another group.")
```

Comment:
Should we put a TODO here? in general it is possible to add quizzes into theory steps